trello-backup.py

```diff
@@ -1,127 +1,173 @@
+                logger.debug("Org Added: {}".format(organization["id"]))
+    return ORGANIZATION_IDS
+
+if __name__ == "__main__":
+
+    # parse args
+    conf_parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        add_help=False)
+    conf_parser.add_argument("-c", "--config", help="Specify Config File")
+
+    args, remaining_args = conf_parser.parse_known_args()
+
+    defaults = {"option": "default"}
+
+    if args.config:
+        config = configparser.SafeConfigParser()
+        config.read([args.config])
+        defaults.update(dict(config.items("Defaults")))
+
+    parser = argparse.ArgumentParser(parents=[conf_parser])
+    parser.set_defaults(**defaults)
+    parser.add_argument("-d", "--output_directory", help="Directory to save JSON files, created if it does not exist")
+    parser.add_argument("-a", "--app_name", help="Trello App Name")
+    parser.add_argument("-k", "--api_key", help="Trello API Key")
+    parser.add_argument("-t", "--token", help="Trello OAuth Token")
+    parser.add_argument("-e", "--token_expiration", help="Trello Token Expiration")
+    parser.add_argument("-l", "--log_level", help="Logging level - info|INFO|WARNING|ERROR|CRITICAL")
+    parser.add_argument("-o", "--organization_ids", help="Organisation by ID(s) to backup. ")
+    parser.add_argument("-n", "--organization_names", help="Organisation by Name(s) to backup")
+    args = parser.parse_args(remaining_args)
+
+    api_url = args.api_url
+    api_key = args.api_key
+    token = args.token
+    token_expiration = args.token_expiration
+    output_directory = args.output_directory
+    app_name = args.app_name
+    if args.organization_ids:
+        ORGANIZATION_IDS = args.organization_ids.lower()
+    if args.organization_names:
+        ORGANIZATION_NAMES = args.organization_names.lower()
+
+    if args.log_level is not None:
+        log_level = args.log_level.upper()
+
+    # Create our logger
+    logger = logging.getLogger("root")
+    logger.setLevel(log_level)
+    ch = logging.StreamHandler()
+    ch.setLevel(log_level)
+    formatter = logging.Formatter("%(levelname)s - %(message)s")
+    ch.setFormatter(formatter)
+    logger.addHandler(ch)
+
+    if not api_key:
+        logger.error("You need an API key to run this app.")
+        logger.error("Visit this url: {}appKey/generate, and re-run script".format(api_url))
+        sys.exit(1)
+
+    if not token:
+        logger.error("You need a token to run this app.")
+        logger.error("Visit this url: {0}connect?key={1}&name={2}&response_type=token&expiration={3}".format(
+            api_url,
+            api_key,
+            app_name,
+            token_expiration))
+        logger.error("Then re-run script")
+        sys.exit(1)
+
+    if ORGANIZATION_NAMES and not ORGANIZATION_IDS:
+        ORGANIZATION_IDS = get_organization_ids(ORGANIZATION_NAMES, api_key, token)
```

Comment:
Its odd that the `ORGANIZATION_IDS` are capitalised to imply a constant, but they are overridden here, perhaps no constants are required because they all appear to be initialized anyway so there are no true constants just globally scoped variables?

It is slightly confusing how we do override these, I think you should try to encapsulate the functionality of getting the ids within the same place, e.g. `get_organization_ids(key, token)` at which time you initialise the names and ids and then filter them accordingly if possible?